xm-commons-tenant/src/main/java/com/icthh/xm/commons/tenant/internal/ThreadLocalAccessorRegistryInterceptor.java

```diff
@@ -0,0 +1,34 @@
+package com.icthh.xm.commons.tenant.internal;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.reflect.Method;
+
+@Slf4j
+public class ThreadLocalAccessorRegistryInterceptor {
+
+    /**
+     * This method registers ThreadLocalAccessor in micrometer ContextRegistry. Ignored if micrometer is missing.
+     * ContextRegistry is used for thread local transfer between threads in Project Reactor.
+     * Please read <a href="https://github.com/micrometer-metrics/context-propagation">context-propagation library</a> for details.
+     *
+     * @param key the key to associate with the ThreadLocal value
+     * @param threadLocal the underlying ThreadLocal
+     */
+    static void register(String key, ThreadLocal<TenantContextDataHolder> threadLocal) {
+        try {
+            Class<?> contextRegistry = Class.forName("io.micrometer.context.ContextRegistry");
```

Comment:
why we using reflection instead of using this class directly?

Reply:
to avoid importing micrometer dependencies
if micrometer added to the classpath and configured by any application this configuration will be applied. If micrometer dependency is missing for your app, this thread local access shearing will be ignored

Reply:
@sergeysenja1992 we took this approach from spring.reactor.core, becasue io.micrometer.context.ContextRegistry available only for reactive spring